main.go

```diff
@@ -243,18 +245,34 @@ func main() {
 
 	policiesLock := &sync.Map{}
 
+	bufferSize := 1024
+
+	replicatedPolicyUpdates := make(chan event.GenericEvent, bufferSize)
```

Comment:
I'm wondering if the buffer size should be dynamic. For example, the number of managed clusters * (number of root policies / 4) or 1024, whichever is larger. The issue is that it can't be resized after it's created.

Reply:
I had roughly the same thoughts... it would be possible to expose it through a flag, which could allow customers to tune it in case we recognize it as a problem in some environments.

From what I could tell in the controller-runtime code, the `replicatedPolicyUpdates` channel here seems to quickly have its items passed into another channel which the controller-runtime creates, which is configured with that `DestBufferSize`. So the current configuration probably actually allows for 2048 policies to be queued in those, before any predicates have even run. (Side note, it looks like only one predicate function per controller is run at a time.)